cpp/src/io/orc/dict_enc.cu

```diff
@@ -113,20 +112,17 @@ static __device__ void LoadNonNullIndices(volatile dictinit_state_s* s,
 
 /**
  * @brief Gather all non-NULL string rows and compute total character data size
- *
- * @param[in] chunks DictionaryChunk device array [rowgroup][column]
- * @param[in] num_columns Number of string columns
  */
 // blockDim {block_size,1,1}
 template <int block_size>
 __global__ void __launch_bounds__(block_size, 2)
   gpuInitDictionaryIndices(DictionaryChunk* chunks,
```

Comment:
Agreed, it would be better. However, the dictionary part of the writer has not yet been "spanified". I'd rather switch all dictionary pointers to spans in one go, and that change is rather large to be added to this PR. Let me know if ti's okay to leave this for a separate PR.

Reply:
If it adds too much burden, I t would be ok to push it as another PR